spanner/value.go

```diff
@@ -976,6 +976,18 @@ func (n *NullProtoEnum) UnmarshalJSON(payload []byte) error {
 	return nil
 }
 
+// NullProtoMessageArray represents an Array of Cloud Spanner PROTO that may be NULL.
+type NullProtoMessageArray struct {
```

Comment:
I cannot find a Null*Array struct for other types. How is proto/enum different?

Reply:
@Shua1 
For other types ex: `NullFloat64`, we can declare an array as `[]NullFloat64` and we know that all elements in the array are of type `Float64`. However, if we consider `[]NullProtomessage`, then `proto.Message` elements in this array can be of any user defined type (in our case `pb.SingerInfo{}`).

If a user just passes an `[]NullProtoMessage` during read operation then we do not know what is the user defined type here (`pb.SingerInfo`), this information is not available on client library side. In order to get this information I have added a Null*Array types, where `ProtoMessageValArray` has the array data, and `ProtoMessageInstance` will have the default instance of user defined type. This `ProtoMessageInstance` field should be passed by user during `write` (to get protoTypefqn) and `read` (to unmarshal data).

Other types do not have this requirement and hence we need to handle it this way for `proto` and `enum`.